source/credentials_provider_sts_web_identity.c

```diff
@@ -564,102 +607,98 @@ static int s_make_sts_web_identity_http_query(
 
 on_error:
     impl->function_table->aws_http_stream_release(stream);
+    aws_input_stream_destroy(input_stream);
     aws_http_message_destroy(request);
 
     return AWS_OP_ERR;
 }
 
-static void s_query_credentials(struct sts_web_identity_user_data *sts_web_identity_user_data) {
-    AWS_FATAL_ASSERT(sts_web_identity_user_data->connection);
+static void s_query_credentials(struct sts_web_identity_user_data *user_data) {
+    AWS_FATAL_ASSERT(user_data->connection);
 
-    struct aws_credentials_provider_sts_web_identity_impl *impl =
-        sts_web_identity_user_data->sts_web_identity_provider->impl;
+    struct aws_credentials_provider_sts_web_identity_impl *impl = user_data->sts_web_identity_provider->impl;
 
     /* "Clear" the result */
-    s_user_data_reset_response(sts_web_identity_user_data);
+    s_user_data_reset_request_and_response(user_data);
 
     /*
-     * Calculate query string:
-     * "/?Action=AssumeRoleWithWebIdentity"
+     * Calculate body message:
```

Comment:
Is there documentation or a link on this data and its format?

Reply:
https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html
There is an example at the bottom of the page, but the data is on query uri, also some of the fields are not required.

Reply:
Can we add a documentation link here then?